tests/functional/iot/test_outfile.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python
+# Copyright 2015 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from awscli.testutils import BaseAWSCommandParamsTest
+import os
+
+
+class TestOutFileQueryArguments(BaseAWSCommandParamsTest):
+    def getpath(self, filename):
```

Comment:
I would take a look at the ``FileCreator`` object from ``testutils``. It is pretty handy when it comes to creating and cleaning files up on the fly. I think the ``FileCreator.rootdir`` will be useful for you because it will get wiped out at the end of each test.